flocker/control/test/test_model.py

```diff
@@ -205,6 +249,24 @@ class NodeStateTests(SynchronousTestCase):
     """
     Tests for ``NodeState``.
     """
+    def test_no_uuid(self):
+        """
+        If no UUID is given, a UUID is generated from the hostname.
+
+        This is done for backwards compatibility with existing tests, and
+        should be removed enveutally.
+        """
+        node = NodeState(hostname=u'1.2.3.4')
+        self.assertIsInstance(node.uuid, UUID)
+
+    def test_uuid(self):
+        """
+        ``NodeState`` can be created with a UUID.
+        """
+        uuid = uuid4()
+        node = NodeState(hostname=u'1.2.3.4', uuid=uuid)
+        self.assertEqual(node.uuid, uuid)
+
```

Comment:
Might be changing again, so I'd like to wait until next branch to see what final interactions look like (eventually `Node` oughtn't accept a hostname).